nucypher/policy/conditions/evm.py

```diff
@@ -187,19 +219,30 @@ def _execute_call(self, parameters: List[Any]) -> Any:
         rpc_result = rpc_function(*parameters)  # RPC read
         return rpc_result
 
-    def verify(self, providers: Dict[int, BaseProvider], **context) -> Tuple[bool, Any]:
+    def verify(
+        self, providers: Dict[int, Set[HTTPProvider]], **context
+    ) -> Tuple[bool, Any]:
         """
         Verifies the onchain condition is met by performing a
         read operation and evaluating the return value test.
         """
-        self._configure_provider(providers=providers)
-        parameters, return_value_test = _resolve_any_context_variables(
-            self.parameters, self.return_value_test, **context
-        )
-        try:
-            result = self._execute_call(parameters=parameters)
-        except Exception as e:
-            raise RPCExecutionFailed(f"Contract call '{self.method}' failed: {e}")
+        for provider in self._next_endpoint(providers=providers):
+            self._configure_provider(provider=provider)
```

Comment:
Should we put the `_configure_provider()` call into `_next_endpoint()`, or is there another reason to keep it out of there? Seems like you'll always need the provider to be configured before using.